src/logic/schedule-parser/workers-info.parser.ts

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { InputFileErrorCode, ScheduleError } from "../../common-models/schedule-error.model";
+import {
+  ContractType,
+  ContractTypeHelper,
+  WorkerAllInfoModel,
+  WorkerType,
+  WorkerTypeHelper,
+} from "../../common-models/worker-info.model";
+import { StringHelper } from "../../helpers/string.helper";
+
+export const DEFAULT_WORKER_TYPE = WorkerType.OTHER;
+export const DEFAULT_CONTRACT_TYPE = ContractType.CIVIL_CONTRACT;
+export const DEFAULT_TIME = 1;
+
+export class WorkersInfoParser {
+  private _workerInfoRows: { [key: string]: WorkerAllInfoModel } = {};
+  private _parseErrors: ScheduleError[] = [];
+
+  constructor(data: string[][]) {
+    this.myWorker(data).forEach((worker) => {
+      this._workerInfoRows[worker.name] = worker;
+    });
+  }
+
+  public get errors(): ScheduleError[] {
+    return [...this._parseErrors];
+  }
+
+  private logLoadFileError(msg: string): void {
+    this._parseErrors.push({
+      kind: InputFileErrorCode.LOAD_FILE_ERROR,
+      message: msg,
+    });
+  }
+
+  private myWorker(raw: string[][]): WorkerAllInfoModel[] {
```

Comment:
Może zmienić na `asWorkerAllInfoModel`, ablo `mapToWorkerAllInfoModel`?

Reply:
W sumie nieważne, ale można by było to zrobić jako funkcję mapuącą. 
W sensie, że dostaje `string[]`, i zwraca `WorkerInfoModel`.

I wtedy nie był potrzbny ten `forEach`.

Po prostu w konstruktorze:
`sectionData = data.map(asWorkerAllInfoModel).forEach(worker=> {this._workerInfoRows[worker.name] = worker})

Reply:
przerobiłam na mapper, a potem wróciłam, 
1) aby zachować konwencję,
2) teraz funkcja zapewnia także filtrację, w mapperze trzebaby było osobno filtrować
sprowadza się to do tego czy chcemy filtrować jawnie

Reply:
Ok